apiserver/src/main/java/org/dependencytrack/util/CsafUtil.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * This file is part of Dependency-Track.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.dependencytrack.util;
+
+import org.apache.commons.validator.routines.DomainValidator;
+import org.apache.commons.validator.routines.UrlValidator;
+
+public final class CsafUtil {
+    private CsafUtil() {}
+
+    public static boolean validateUrl(String url) {
+        return UrlValidator.getInstance().isValid(url);
+    }
+
+    public static boolean validateDomain(String domain) {
+        return DomainValidator.getInstance().isValid(domain);
```

Comment:
Can we find a way to do this without requiring an additional dependency?

Reply:
I did a very crude implementation of a domain validator and also moved it to the `.validation` package.